src/kmeans.jl

```diff
@@ -116,35 +146,32 @@ function _kmeans!(
         else
             # if only a small subset is affected, only compute for that subset
             affected_inds = findall(to_update)
-            dmat_p = pairwise(distance, centers[:, affected_inds], X)
-            dmat[affected_inds, :] .= dmat_p
+            pairwise!(view(dmat, affected_inds, :), distance,
+                      view(centers, :, affected_inds), X)
         end
 
         # update assignments
-
-        update_assignments!(dmat, false, assignments, costs, counts, to_update, unused)
-
+        update_assignments!(dmat, false, assignments, costs, counts,
+                            to_update, unused)
         num_affected = sum(to_update) + length(unused)
 
         # compute change of objective and determine convergence
         prev_objv = objv
-        objv = w === nothing ? sum(costs) : dot(w, costs)
+        objv = weights === nothing ? sum(costs) : dot(weights, costs)
```

Comment:
I don't see the point of renaming this. Please avoid unnecessary changes as it makes the PR very hard to review.

Reply:
Yeah I don't want to be that guy, however I did try to pick consistent names throughout so that there's no function that calls weights at the top (`kmeans` and `kmeans!`) which then goes on to be called  `w`. I think it'll make further iteration of improvements easier if we stick to the same names for the same things?